src/plugins/intel_gpu/src/plugin/compiled_model.cpp

```diff
@@ -35,9 +35,27 @@ std::shared_ptr<ov::threading::ITaskExecutor> create_task_executor(const std::sh
     if (config.get_property(ov::internal::exclusive_async_requests)) {
         //exclusive_async_requests essentially disables the streams (and hence should be checked first) => aligned with the CPU behavior
         return plugin->get_executor_manager()->get_executor("GPU");
-    } else {
+    } else if (config.get_property(ov::num_streams) > 1) {
```

Comment:
CPU pinning will work for all number of streams. Please remove `if (config.get_property(ov::num_streams) > 1)`

Reply:
Updated.